overrides/404.html

```diff
@@ -0,0 +1,26 @@
+{% extends "main.html" %}
+
+{% block content %}
+<script>
+    function toLowerCaseExceptHash(url) {
+        var urlParts = url.split('#');
+        urlParts[0] = urlParts[0].toLowerCase();
+        return urlParts.join('#');
+    }
+
+    var currentUrl = window.location.href;
+    var lowerCaseUrl = toLowerCaseExceptHash(currentUrl);
+
+    if (currentUrl !== lowerCaseUrl) {
+        document.write("<h1>Redirecting...</h1>");
+        window.location.href = lowerCaseUrl;
+
+    } else {
+        document.write("<h1>404 - Page not found</h1>");
+        document.write(
+            "<p>It looks like the requested page doesn't exist! You may be able to find the right information via our ",
+            "search function.</p>"
+        );
+    }
+</script>
```

Comment:
You could add a notification here in HTML for users that have javascript disabled. And maybe replace the innex text with the message you want
'''
let h1 = document.querySelector("h1")
h1.innerText = "Correct Text"
'''

Reply:
@berrydenhartog LGTM, I made some changes based on your suggestions. Thx.